azure/datalake/store/lib.py

```diff
@@ -214,9 +248,9 @@ def session(self):
         return s
 
     def _check_token(self):
-        if time.time() - self.token['time'] > self.token['expiresIn'] - 100:
-            self.token = refresh_token(self.token)
-            self.head = {'Authorization': 'Bearer ' + self.token['access']}
+        cur_session = self.token.signed_session()
+        if not self.head or not 'Authorization' in self.head or self.head['Authorization'] != cur_session.headers['Authorization']:
```

Comment:
Could possibly simplify this to `if not self.head or self.head.get('Authorization') != cur_session.headers['Authorization']`.